tests/containers/enable_mac.pm

```diff
@@ -0,0 +1,92 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Package: containers
+# Summary: Enable SELinux or AppArmor
+#
+# Maintainer: QA-C team <qa-c@suse.de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use Utils::Systemd qw(systemctl);
+use Utils::Logging 'save_and_upload_log';
+use bootloader_setup qw(replace_grub_cmdline_settings);
+use power_action_utils qw(power_action);
+use utils qw(zypper_call);
+use transactional qw(trup_call check_reboot_changes process_reboot);
+use version_utils qw(is_transactional);
+
+# Taken from tests/transactional/enable_selinux.pm
+sub check_enforcing {
+    assert_script_run('selinuxenabled');
+    validate_script_output("getenforce", sub { m/Enforcing/ });
+    validate_script_output("sestatus", sub { m/Current mode:.*enforcing/ });
+    validate_script_output("sestatus", sub { m/Mode from config file:.*enforcing/ });
+    record_info('SELinux', script_output('sestatus'));
+    record_info('Audit report', script_output('aureport'));
+    record_info('Audit denials', script_output('aureport -a', proceed_on_failure => 1));
+}
+
+sub run {
+    my ($self) = @_;
+
+    if (is_transactional) {
+        select_serial_terminal;
+    } else {
+        select_console 'root-console';
+    }
+
+    my $current_mac = script_output("grep -Eo '(selinux|apparmor)' /sys/kernel/security/lsm");
+    my $security_mac = get_required_var("SECURITY_MAC");
+    die "Invalid value for SECURITY_MAC: $security_mac" unless ($security_mac =~ /selinux|apparmor/);
+
+    return if ($security_mac eq $current_mac);
+    record_info "SECURITY_MAC", "Switching from $current_mac to $security_mac";
+
+    unless (is_transactional) {
+        zypper_call "rm -t pattern $current_mac";
+        zypper_call "in -t pattern $security_mac";
+    }
+
+    if ($security_mac eq "selinux") {
+        if (is_transactional) {
+            # NOTE: There's no setup-apparmor, so this test scenario is not tested in this module
+            trup_call('setup-selinux');
+        } else {
+            replace_grub_cmdline_settings('security=apparmor', 'security=selinux selinux=1', update_grub => 1);
+            assert_script_run "sed -i -e 's/^SELINUX=.*/SELINUX=enforcing/g' /etc/selinux/config";
+            assert_script_run "touch /.autorelabel";
+        }
+    } else {
+        replace_grub_cmdline_settings('security=selinux selinux=1', 'security=apparmor', update_grub => 1);
+    }
+
+    if (is_transactional) {
+        process_reboot(trigger => 1);
+    } else {
+        power_action('reboot', textmode => 1);
+        $self->wait_boot(bootloader_time => 300);
+        select_serial_terminal;
```

Comment:
Maybe this can be moved after the if-else so the test enforces using serial terminal